src/views/transaction-detail/index.js

```diff
@@ -142,11 +141,11 @@ class TransactionDetail extends Component {
         }
 
         // Retrieve the tx and txhash from state
-        const {txHash} = this.state;
+        const { txHash, loadFromCache } = this.state;
 
-        if (txHash !== undefined) {
+        if (txHash !== undefined && !loadFromCache) {
             // Retrieve the tx information from the network
-            this.getTx(txHash.txHash);
+            this.getTx(txHash);
```

Comment:
Interesting—shouldn't we do this asynchronously?

EDIT: Ah, interesting, I think this is okay.

Reply:
Not my doing per se, just a fix of what was there already